fdbserver/worker.actor.cpp

```diff
@@ -448,6 +451,22 @@ ACTOR Future<Void> runProfiler(ProfilerRequest req) {
 	}
 }
 
+void runHeapProfiler() {
+#if defined(__linux__) && defined(USE_GPERFTOOLS) && !defined(VALGRIND)
+		HeapProfilerDump("User triggered heap dump");
```

Comment:
You might wish to look into if there's an error that you can throw here, that would bubble back into the CLI to indicate that you requested a heap profile, but none was taken because the code wasn't enabled.  `client_invalid_operation()` ?

Reply:
I agree an error sounds nice, but `client_invalid_operation` may imply that the person typing the command did something wrong. There's another called `unsupported_operation` which may work, or we could create a new one if that seemed desirable.

Reply:
[HeapProfilerDump()](https://github.com/gperftools/gperftools/blob/master/src/gperftools/heap-profiler.h#L92) returns void and there is no error code for checking. Finding the dump'ed file is tricky -- the file name is determined by the tcmalloc library in the format of "XXX.000Y.heap", where XXX is file prefix and Y is a sequence number.

So I only checked if heap profiling is enabled here.

Reply:
Discussed offline with @alexmiller-apple, throwing an error doesn't work because there is no one waiting on the caller `uncancellable(runProfiler(profilerReq))`. This is intentional so that `fdbcli` doesn't wait for the duration of profiling. So I just added a trace event in the failed cases.